### DISCUSSION TITLE
Code formatting, config

### DISCUSSION THREAD

**Discussion by hmpf** [Uninett/Howitz#117] [General]
So we're planning to run [ruff check --fix](https://docs.astral.sh/ruff/) on `main` and all branches. That means we need to settle on some code style things. ([Docs for all ruff settings](https://docs.astral.sh/ruff/settings/).)

Looking at the [overview docs for ruff formatting config](https://docs.astral.sh/ruff/formatter/#configuration):

```
[tool.ruff]
line-length = 100

[tool.ruff.format]
quote-style = "single"
indent-style = "tab"
docstring-code-format = true
```

Defaults:
```
[tool.ruff]
line-length = 88

[tool.ruff.format]
quote-style = "double"
indent-style = "space"
indent-width = 4
docstring-code-format = false
```

Homework: read through the settings docs and see if there's more that would be nice to tweak away from the default.

**Comment by hmpf** [comment]
1. Let's keep it in pyproject.toml
2. `indent-style` should be "space", non-negotiable
3. Let's wait with choosing a docstyle-format, we're not good enough with adding docstrings as it is
4. I prefer single quotes if I have to choose, so `quote-style = "single"`
5. Line-length, I prefer shorter more than longer

**Reply by podliashanyk** [reply]
All sound good to me

**Comment by hmpf** [comment]
I'd like to turn off complaints about annotations, or at least set

```
[tool.ruff.lint.flake8-annotations]
allow-star-arg-any = true
ignore-fully-untyped = true
suppress-dummy-args = true
suppress-none-returning = true
```

**Comment by hmpf** [comment]
Here's more:

```
[tool.ruff.lint.flake8-unused-arguments]
ignore-variadic-names = true
```

**Comment by hmpf** [comment]
Should we sort imports? Lots of flags for that: [Config for sorting imports](https://docs.astral.sh/ruff/settings/#lintisort)

**Reply by podliashanyk** [reply]
Yes please

**Comment by podliashanyk** [comment]
> Homework: read through the settings docs and see if there's more that would be nice to tweak away from the default.

1. Add `exclude` and/or `respect-gitignore`? 
```toml
[tool.ruff]
exclude = [".venv"]
respect-gitignore = true
```
